plugins/ingest/src/main/java/org/elasticsearch/ingest/Pipeline.java

```diff
@@ -65,7 +65,7 @@ public Builder(String name) {
             this.name = name;
         }
 
-        public Builder(Map<String, Object> config) {
+        public Builder(Map<String, Object> config, Map<String, Processor.Builder> processorRegistry) {
```

Comment:
@uboness makes sense, I added the factory for the processor builders, so the registry passed to the processor builder doesn't hold state any more.